### DISCUSSION TITLE
Prevent Uploading Same File (or fileName) Multiple Times in FileUpload Component

### DISCUSSION THREAD

**Discussion by elson-currentcorp** [chakra-ui/zag#2025] [Ideas]
It would be nice to introduce a new error type, or prop for rejecting an uploaded file through the FileUpload component if the same file (or file name) already exists.

**Comment by anubra266** [comment]
@elson-currentcorp you can use `validate` method.
by using some state or ref to hold the acceptedFiles or rejectedFiles.
Validate method would look like this:  
![CleanShot 2024-11-20 at 22 32 26@2x](https://github.com/user-attachments/assets/29c0ca26-577c-44bd-ae34-f1aaea0eea98)

**Reply by elson-currentcorp** [reply]
Didn't know you could set a custom error with validate.

**Reply by anubra266** [reply]
@elson-currentcorp  We will update the docs to make this clear

**Reply by assaftenen** [reply]
any updates?

**Reply by anubra266** [reply]
@assaftenen updates on what?

**Reply by assaftenen** [reply]
@anubra266 thanks for the quick reply.
I meant for the docs update.. and I found it - thanks!!!
